classes/ffprobe.php

```diff
@@ -145,12 +145,31 @@ public function get_media_metadata(\stored_file $file) : array {
 
         // Execute the FFProbe command to get file metadata.
         $command = $this->ffprobe_path . ' -of json -v error -show_format -show_streams ' .  escapeshellarg($tempfile);
-        $rawresults = shell_exec($command);
+        $errfile = $tempfile . "_err";
+        // Send stderr to the err file to retrieve seperate from stdout.
+        $rawresults = shell_exec("$command 2>$errfile");
         unlink($tempfile); // Remove temp file.
 
-        if ($rawresults) {
+        // Check status of errors and results.
+        $exit = false;
+        $errs = file_get_contents($errfile);
+        unlink($errfile);
+        if (!empty($errs)) {
+            $exit = true;
+        }
+
+        if ($rawresults) { // We got a result, check it for sanity
             $resultobject = json_decode($rawresults);
+            if ($resultobject == []) {
+                // FFprobe must have errored and returned empty JSON.
```

Comment:
is there also a case where $errs is empty, but the command returned something which wasn't json? in which case maybe dump the malformed json into the $errs. might never happen but can't hurt